tests/tests/mempool.rs

```diff
@@ -0,0 +1,396 @@
+    // Check if the retrieved hash matches the expected hash
+    assert_eq!(hashes.unwrap().hash(), &transaction_signed.hash());
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_mempool_add_transactions(#[future] katana: Katana, _setup: ()) {
+    let eth_provider = katana.eth_provider();
+    // Get the EOA address
+    let address = katana.eoa().evm_address().expect("Failed to get eoa address");
+
+    // Set the number of transactions to create
+    let transaction_number = 2;
+
+    // Create multiple sample transactions
+    let transactions =
+        create_sample_transactions(&katana, transaction_number).await.expect("Failed to create sample transaction");
+
+    // Collect pooled transactions
+    let pooled_transactions =
+        transactions.iter().map(|(eth_pooled_transaction, _)| eth_pooled_transaction.clone()).collect::<Vec<_>>();
+
+    // Collect signed transactions
+    let signed_transactions =
+        transactions.iter().map(|(_, signed_transactions)| signed_transactions.clone()).collect::<Vec<_>>();
+
+    // Add transactions to mempool
+    let _ = eth_provider.mempool().unwrap().add_transactions(TransactionOrigin::Local, pooled_transactions).await;
+
+    // Get pending transactions
+    let hashes = eth_provider.mempool().unwrap().pending_transactions();
+
+    // Get transactions by sender address
+    let sender_transactions = eth_provider.mempool().unwrap().get_transactions_by_sender(address);
+
+    // Get unique senders from the mempool
+    let unique_senders = eth_provider.mempool().unwrap().unique_senders();
+
+    // Check if the first signed transaction is contained
+    let contains = eth_provider.mempool().unwrap().contains(&signed_transactions[0].hash());
+
+    // Get updated mempool size
+    let mempool_size = eth_provider.mempool().unwrap().pool_size();
+    // mempool_size function tests
+    // Check pending transactions
+    assert_eq!(mempool_size.pending, transaction_number);
+    // Check queued transactions
+    assert_eq!(mempool_size.queued, 0);
+    // Check total transactions
+    assert_eq!(mempool_size.total, transaction_number);
+
+    // pending_transactions function tests
+    // Check if the first pending transaction hash matches
+    assert_eq!(hashes[0].hash(), &signed_transactions[0].hash());
+    // Check if the second pending transaction hash matches
+    assert_eq!(hashes[1].hash(), &signed_transactions[1].hash());
+    // Ensure the number of pending transactions matches the expected count
+    assert_eq!(hashes.len(), transaction_number);
+
+    // get_transactions_by_sender function tests
+    // Ensure only one transaction is returned
+    assert_eq!(sender_transactions.len(), transaction_number);
+    // Check if the returned transaction hash matches
+    assert_eq!(*sender_transactions[0].hash(), signed_transactions[0].hash());
+    assert_eq!(*sender_transactions[1].hash(), signed_transactions[1].hash());
```

Comment:
can't we replace all this by retrieving expected_hashes in a vec and received hashes in another vec?

Reply:
also I think it's nicer when the assert follows the actual function call directly, it's easier to follow the test.
So something like
```rust
    // Get pending transactions
    let hashes = eth_provider.mempool().unwrap().pending_transactions();
    // Check if the first pending transaction hash matches
    assert_eq!(hashes[0].hash(), &signed_transactions[0].hash());
    // Check if the second pending transaction hash matches
    assert_eq!(hashes[1].hash(), &signed_transactions[1].hash());
    // Ensure the number of pending transactions matches the expected count
    assert_eq!(hashes.len(), transaction_number);```

Reply:
fixed on all the tests, thanks